src/webcomponents/organization/admin/user-admin-update.js

```diff
@@ -0,0 +1,344 @@
+                type: String,
+            },
+            active: {
+                type: Boolean,
+            },
+            opencgaSession: {
+                type: Object
+            },
+            displayConfig: {
+                type: Object
+            },
+        };
+    }
+
+    #init() {
+        this._user = {};
+        this.userId = "";
+        this.studyId = "";
+        this.displayConfig = {};
+        this.updatedFields = {};
+
+        this._config = this.getDefaultConfig();
+    }
+
+    #setLoading(value) {
+        this.isLoading = value;
+        this.requestUpdate();
+    }
+
+    #initConfigNotification() {
+        this._config.notification = {
+            title: "",
+            text: "Some changes have been done in the form. Not saved changes will be lost",
+            type: "notification",
+            display: {
+                visible: () => {
+                    return UtilsNew.isNotEmpty(this.updatedFields);
+                },
+                notificationType: "warning",
+            },
+        };
+    }
+
+    #initUser() {
+        // 1. Group contains params: (a) id: e.g. "@admins", (b) userIds: e.g. ["test"]
+        this._user = UtilsNew.objectClone(this.user);
+        // 2. In the update form, we need to manage as well the permissions of this group.
+        // Retrieve ACL permissions. Check if this study group has acl
+        // CAUTION: study does not have acl?
+        // const groupPermissions = this._study?.acl
+        //     ?.find(acl => acl.member === this.opencgaSession.user.id)?.groups
+        //     ?.find(group => group.id === this.group.id)?.permissions || [];
+        // // 3. Add current permissions and template key to the object group
+        // this.group = {
+        //     ...this.group,
+        //     permissions: groupPermissions,
+        //     template: "", // Fixme: not sure how to retrieve template
+        // };
+        this.initOriginalObjects();
+    }
+
+    #initOriginalObjects() {
+        this._user = UtilsNew.objectClone(this.user);
+        this.updatedFields = {};
+    }
+
+    update(changedProperties) {
+        if ((changedProperties.has("userId") && this.active)) {
+            this.userIdObserver();
+        }
+        if (changedProperties.has("displayConfig")) {
+            this._config = this.getDefaultConfig();
+            if (!this._config?.notification) {
+                this.#initConfigNotification();
+            }
+        }
+        super.update(changedProperties);
+    }
+
+    userIdObserver() {
+        if (this.userId && this.opencgaSession) {
+            const params = {
+                organization: this.organization.id,
+            };
+            let error;
+            this.#setLoading(true);
+            this.opencgaSession.opencgaClient.users()
+                .info(this.userId, params)
+                .then(response => {
+                    this.user = UtilsNew.objectClone(response.responses[0].results[0]);
```

Comment:
Do you really need to generate a clone of the result returned by OpenCGA?

Reply:
Not really, just followed the code in opencga-update.js. I'll make the necessary changes in the next task.